File: react_todo_frontend/src/domain/value-object/TitleVO.ts
//バリデーション
export const TitleVO = (title: string | null): string => {
    if (title === null || title.trim().length === 0 || title.length > 50) {
        throw new Error("タイトルは1文字以上50文字以下で入力してください");
    }
    return title; //エラーなし
}

File: go_todo_api/domain/vo/title_vo.go
package value_object

import (
	"errors"
	"strings"
)

type Title struct {
	value string
}

func FromStringTitle(v string) (Title, error) {
	trimmed := strings.TrimSpace(v)

	if len(trimmed) == 0 || len(trimmed) > 50 {
		return Title{}, errors.New("タイトルは1文字以上50文字以下である必要があります")
	}

	return Title{value: trimmed}, nil
}

// Titleの値を取得するメソッド
func (t Title) Value() string {
	return t.value
}
